docs/source/notebooks/wps_generate_climos.ipynb

```diff
@@ -0,0 +1,388 @@
+    "generate_climos:\n",
+    "INFO:dp.generate_climos:Processing: https://docker-dev03.pcic.uvic.ca/twitcher/ows/proxy/thredds/dodsC/datasets/TestData/fdd_seasonal_CanESM2_rcp85_r1i1p1_1951-2100.nc\n",
+    "INFO:dp.generate_climos:climo_periods: {'6190'}\n",
+    "INFO:dp.generate_climos:project: CMIP5\n",
+    "INFO:dp.generate_climos:institution: PCIC\n",
+    "INFO:dp.generate_climos:model: CanESM2\n",
+    "INFO:dp.generate_climos:emissions: historical, rcp85\n",
+    "INFO:dp.generate_climos:run: r1i1p1\n",
+    "INFO:dp.generate_climos:dependent_varnames: ['fdd']\n",
+    "INFO:dp.generate_climos:time_resolution: seasonal\n",
+    "INFO:dp.generate_climos:is_multi_year_mean: False\\n'''\n",
+    "\n",
+    "assert output_data == expected_output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# generate climos\n",
+    "output = thunderbird.generate_climos(\n",
+    "    netcdf=seasonal_opendap, \n",
+    "    operation=operation, \n",
+    "    climo=climo, \n",
+    "    resolutions=resolutions, \n",
+    "    dry_run=False\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<metaurl mediatype=\"application/x-netcdf\">\n",
+      "     http://localhost:5001/outputs/5657bfa8-0368-11eb-b7e1-c9fa1b0c3983/fdd_aClimMean_BCCAQ_CanESM2_historical+rcp85_r1i1p1_19610101-19901231_Canada.nc\n",
+      "    </metaurl>\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Process normal output\n",
+    "req = requests.get(output.get()[0])\n",
+    "soup = BeautifulSoup(BeautifulSoup(req.content.decode('utf-8')).prettify()).metaurl\n",
+    "print(soup)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "<ipython-input-9-e3c8cb77b746>:9: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.\n",
+      "  for key in input_data.__dict__:\n",
+      "<ipython-input-9-e3c8cb77b746>:12: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.\n",
+      "  assert input_data.__dict__[key] == output_data.__dict__[key]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Test normal output data is the same as the input data\n",
+    "input_data = Dataset(seasonal_opendap)\n",
+    "url = soup.get_text()\n",
+    "\n",
+    "with NamedTemporaryFile(suffix=\".nc\", prefix=\"tmp_copy\", dir=\"/tmp\", delete=True) as tmp_file:\n",
+    "    urlretrieve(url, tmp_file.name)\n",
+    "    output_data = Dataset(tmp_file.name)\n",
+    "\n",
+    "for key in input_data.__dict__:\n",
+    "    # History and frequency are different\n",
+    "    if key != 'frequency' and key != 'history':\n",
+    "        assert input_data.__dict__[key] == output_data.__dict__[key]"
```

Comment:
Given that the purpose of  this process is to transform the data, we may want to focus on testing differences rather than asserting commonalities. In this case we could be checking that `frequency` and `history` are in fact different, rather than leaving them out.

Another simple test could be to check the dimensions. Creating a 30 year average with the given parameters will squash the time dimension down to `1` from `x` (where `x` is the number of days, I think in this case it is `600`) and we can look for that, or just that the datasets are now different
```
assert input_data.dimensions == output_data.dimensions
```